mne/beamformer/_lcmv.py

```diff
@@ -39,28 +39,36 @@ def _reg_pinv(x, reg):
     return linalg.pinv(x), d
 
 
-def _setup_picks(picks, info, forward, noise_cov=None):
-    if picks is None:
-        picks = pick_types(info, meg=True, eeg=True, ref_meg=False,
-                           exclude='bads')
-
-    ok_ch_names = set([c['ch_name'] for c in forward['info']['chs']])
+def _setup_picks(info, forward, data_cov=None, noise_cov=None):
+    # get a list of all channel names:
+    fwd_ch_names = set([c['ch_name'] for c in forward['info']['chs']])
+
+    # handle channels from forward model and info:
+    ch_names = [c['ch_name'] for c in info['chs']
+                if c['ch_name'] not in info['bads'] and
+                c['ch_name'] in fwd_ch_names]
+
+    # handle channels from data cov if data cov is not None
+    # Note: data cov is supposed to be None in tf_lcmv
+    if data_cov is not None:
+        ch_names = [c for c in ch_names
+                    if c not in data_cov['bads'] and
+                    c in data_cov.ch_names]
+
+    # handle channels from noise cov if noise cov available:
     if noise_cov is not None:
-        ok_ch_names.union(set(noise_cov.ch_names))
+        ch_names = [c for c in ch_names
+                    if c not in noise_cov['bads'] and
+                    c in noise_cov.ch_names]
 
+    # inform about excluding channels:
+    if data_cov is not None and set(info['bads']) != set(data_cov['bads']):
+        logger.info('info["bads"] and data_cov["bads"] do not match, '
+                    'excluding bad channels from both.')
```

Comment:
That's been there already :wink: - but actually, I wonder if we should change it. It gives you the message about excluding channels even if there is nothing excluded - e.g., when the data or noise cov contain an EEG channel marked as bad but EEG channels are not selected anyway in ``info``. This can be misleading, right?